skills/record-plan/scripts/check_lock.py

```
@@ -0,0 +1,38 @@
+
+import os
+import sys
+
+def check_lock(working_dir: str, file_path: str) -> bool:
+    """Check if lock file exists and authorizes this file."""
+    lock_file = os.path.join(working_dir, '.record-plan-active')
+
+    if not os.path.exists(lock_file):
+        print("❌ ERROR: No active record-plan session")
+        print("MUST invoke wrapper first:")
+        print(f"  python3 ~/.claude/skills/record-plan/scripts/write_plan.py \\")
```

Ruff (0.14.8)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)